protocol/proofs/delayed-weth-expansion.md

```diff
@@ -0,0 +1,62 @@
+
+## Problem Statement
+
+Existing security mechanisms within the `DelayedWETH` contract are inconvenient and slow down the
+incident response process. Specifically:
+
+- The `DelayedWETH.hold(address target)` function creates an approval from the `target` address to
+  the `owner` address. Since the `target` can simply remove this approval, `hold` MUST be triggered
+  alongside a `transferFrom` in the same transaction. This adds unnecessary complexity to something
+  that could simply be a single transaction.
+- The various security mechanisms inside of `DelayedWETH` can only be triggered by the System Owner
+  which has a slow SLA. The Deputy Guardian can act to trigger the Superchain-wide pause, but this
+  is highly disruptive to the entire Superchain ecosystem.
+
+Additionally, it should be noted that all ETH inside a `DelayedWETH` contract is pooled together
+regardless of which games are using the contract. We currently mitigate this by using one
+`DelayedWETH` contract per game type, but this expands the number of contracts that the System
+Owner and Guardian roles must manage by at least 2 per additional chain. However, the modifications
+required to appropriately address this issue are sufficiently extensive that they are considered
+out of scope for this proposal.
+
+## Propsed Solution
+
+Here we propose some modifications to `DelayedWETH` that help to resolve the two noted limitations:
+
+1. We propose that `DelayedWETH.hold` executes a `transferFrom` automatically to reduce the number
+  of required transactions from 2 to 1 when holding ETH from a single dispute game.
+2. We propose a new `pause` function on the `DelayedWETH` contract itself that pauses *just* the
+  `DelayedWETH` contract and supplements the Superchain-wide pause functionality. This `pause`
+  function would be executable by the Guardian role and could therefore be triggered quickly in
+  case the System Owner is unable to execute another security action in time.
+
+## Alternatives Considered
+
+### Wider Rewrite
+
+We considered the alternative to rewrite the `DelayedWETH` contract more broadly to be able to
+easily manage multiple game types across multiple chains within a single `DelayedWETH` instance.
+However, this would require significant modifications to the `DelayedWETH` contract and would act
+as an iterative improvement to the status quo. We still believe that a larger rewrite should be
+considered for the near future.
+
+## Risks & Uncertainties
+
+### Security Model
+
+We see little risk with the modification to the `hold` function. Introducing a new `pause` function
+slightly changes the existing security model (though not significantly). The Guardian role can
+already pause the `DelayedWETH` contract by triggering the Superchain-wide pause. Allowing this
+role to pause the `DelayedWETH` contract without pausing the rest of the system should not open up
+capabilities that the Guardian does not already have. Comments on this subject are welcome.
```

Comment:
Regarding the `DelayedWETH.pause` function, this would also require a change to the DeputyGuardianModule (DGM), this way both the Guardian and Deputy Guardian can call `DelayedWETH.pause`. Changing the DGM invalidates all presigned pauses. The Granite changes also changed the DGM and therefore invalidated the pauses, so they were resigned.

Since it seems the DGM changes more frequently than we thought, we should consider proxying as part of this scope, to avoid having to resign presigned pauses with each DGM change. Assuming proxying it is required additional work, does that change the decision on this vs. the alternative rewrite? I'm unsure how large the scope of "Wider Rewrite" is

Reply:
Ah, interesting. Good point. DGM starts to get increasingly bloated, especially if there are a bunch of different `DelayedWETH` contracts all over the place. I still think proxying the DGM would be much easier than rewriting `DelayedWETH` for now. I do think rewriting `DelayedWETH` is a good idea but the goal is to be able to ship these improvements within a few months and a rewrite would take a lot longer.

Reply:
+1 on proxying the DGM. Having to update presigned pauses each time we change it will be a bit painful.

Reply:
Clearly Agreed about with the fact we need to proxify the DGM!  
I would also suggest to inform the reader that this would require to deploy a new DGM.  
```Markdown
### Changes to the DeputyGuardianModule
The `DeputyGuardianModule` will have few light changes but this would require a total redeployment on L1. Thus, breaking the current PSPs and would need to regenerate new PSPs from the new `DeputyGuardianModule`.
```